adapters/rack/lib/opentelemetry/adapters/rack/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,262 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require_relative '../util/queue_time'
+
+module OpenTelemetry
+  module Adapters
+    module Rack
+      module Middlewares
+        # TracerMiddleware propagates context and instruments Rack requests
+        # by way of its middleware system
+        #
+        # Notable implementation differences from dd-trace-rb:
+        # * missing: 'span.resource', which is set to span.name
+        # * missing: config[:distributed_tracing]
+        # * missing: span.set_error() -- spec level change
+        class TracerMiddleware # rubocop:disable Metrics/ClassLength
+          class << self
+            def allowed_rack_request_headers
+              @allowed_rack_request_headers ||= {}.tap do |result|
```

Comment:
This might be cleaner if we: lose the intermediary `allowed_request_header_names` method and use `each_with_object` instead of `{}.tap`. E.g.

```

@allowed_rack_request_headers ||= \
  Array(config[:allowed_request_headers]).each_with_object({}) do |header, memo|
    ...
  end
```